File: backend/src/main/java/com/inventory/backend/repository/ActionLogRepository.java
package com.inventory.backend.repository;

import com.inventory.backend.entity.ActionLog;
import com.inventory.backend.entity.ActionLogType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

import java.time.LocalDateTime;
import java.util.Collection;
import java.util.List;

public interface ActionLogRepository extends JpaRepository<ActionLog, Long>, JpaSpecificationExecutor<ActionLog> {
    List<ActionLog> findByActionTypeInAndCreatedAtBetween(Collection<ActionLogType> actionTypes, LocalDateTime from, LocalDateTime to);
}

File: backend/src/main/kotlin/com/inventory/backend/repository/ActionLogRepository.kt
package com.inventory.backend.repository

import com.inventory.backend.entity.ActionLog
import com.inventory.backend.entity.ActionLogType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import java.time.LocalDateTime

interface ActionLogRepository : JpaRepository<ActionLog, Long>, JpaSpecificationExecutor<ActionLog> {
    fun findByActionTypeInAndCreatedAtBetween(
        actionTypes: Collection<ActionLogType>,
        from: LocalDateTime,
        to: LocalDateTime,
    ): List<ActionLog>
}
